frontend/src/hooks/useLoginState.ts

```diff
@@ -14,6 +15,7 @@ export const useLoginState = () => {
 
   const logout = useCallback(() => {
     dispatch(logoutAction());
+    dispatch(clearSettings());
```

Comment:
It could be good to explain to the other developers the reason behind this clear.

```suggestion
    /**
     * Ensure that the user's settings retrieved after the login are cleared 
     * after a logout, to not leave trace of the user's preferences in the
     * browser. 
     */
     dispatch(clearSettings());
```